diff --git a/Install.html b/Install.html
index 7ef3ec5..dc01184 100644
--- a/Install.html
+++ b/Install.html
@@ -12,8 +12,10 @@
 进行编译之前需安装的几个开发包如下：
 对应Ubuntu/Debian的是：
 	sudo apt-get install libpcap-dev libssl-dev
+	sudo apt-get install pkg-config
 对应Fedora/Redhat的是：
 	yum install libpcap-devel openssl-devel
+	yum install pkgconfig
 
 从源码包开始编译客户端的命令为：
 	tar xzf njit8021xlient-1.1.tar.gz
diff --git a/ReadMe.html b/ReadMe.html
index 0c37f86..53dd65e 100644
--- a/ReadMe.html
+++ b/ReadMe.html
@@ -51,6 +51,9 @@ Wiki文档
 GitHub开源项目托管
  http://github.com/liuqun/njit8021xclient
 
+SourceForge开源项目托管
+ http://sourceforge.net/projects/njit8021xclient/
+
 [致谢]
 感谢编写兼容客户端的先驱者们！
 南京理工大学：AGanNo2、throqq
diff --git a/src-gui/luci/luci_controller_njit.lua b/src-gui/luci/luci_controller_njit.lua
new file mode 100644
index 0000000..cefbfa5
--- /dev/null
+++ b/src-gui/luci/luci_controller_njit.lua
@@ -0,0 +1,5 @@
+module("luci.controller.njit", package.seeall)
+
+function index()
+	entry({"admin", "network", "njit"}, cbi("njit"), _("njit-client"), 88).i18n = "wol"
+end
diff --git a/src-gui/luci/njit.lua b/src-gui/luci/njit.lua
new file mode 100644
index 0000000..89ca878
--- /dev/null
+++ b/src-gui/luci/njit.lua
@@ -0,0 +1,61 @@
+--[[
+LuCI - Lua Configuration Interface
+
+Copyright 2010 Jo-Philipp Wich <xm@subsignal.org>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+]]--
+
+local sys = require "luci.sys"
+local fs  = require "nixio.fs"
+
+m = SimpleForm("njit", translate("njit-client"),
+	translate("njit 802.11x client"))
+
+m.submit = translate("Connect")
+m.reset  = false
+
+r = m:field(Value, "username", translate("Username"))
+s = m:field(Value, "password", translate("Password"))
+t = m:section(SimpleSection)
+interface = t:option(ListValue, "interface", translate("Network interface"))
+for _, e in ipairs(sys.net.devices()) do
+		if e ~= "lo" then interface:value(e) end
+	end
+
+
+function interface.write(self, s, val)
+	local cmd
+	local user = luci.http.formvalue("cbid.njit.1.username")
+	local pass = luci.http.formvalue("cbid.njit.1.password")
+	local iface = luci.http.formvalue("cbid.njit.1.interface")
+	cmd = "njit-client %s %s %s & (sleep 10 && killall njit-client)" %{ user, pass, iface }
+
+	local msg = "<p><strong>%s</strong><br /><br /><code>%s<br /><br />" %{
+		translate("Starting njit-client:"), cmd
+		}
+
+	local p = io.popen(cmd .. " 2>&1")
+	if p then
+		while true do
+			local l = p:read("*l")
+			if l then
+				if #l > 100 then l = l:sub(1, 100) .. "..." end
+				msg = msg .. l .. "<br />"
+			else
+				break
+			end
+		end
+		p:close()
+	end
+	msg = msg .. "</code></p>"
+	m.message = msg
+end
+
+return m
+
+
diff --git a/src-gui/luci/readme b/src-gui/luci/readme
new file mode 100644
index 0000000..b995d16
--- /dev/null
+++ b/src-gui/luci/readme
@@ -0,0 +1,2 @@
+cp ./njit.lua /usr/lib/lua/luci/model/cbi/njit.lua
+cp ./luci_controller_njit.lua /usr/lib/lua/luci/controller/njit.lua
