diff --git a/src/Makefile.am b/src/Makefile.am
index 8363241..93ebfdb 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -10,14 +10,11 @@ client_SOURCES = \
 		njit8021xclient.c \
 		njit8021xclient.h \
 		debug.h \
-		md5-buildin/md5_one.c \
-		md5-buildin/md5_dgst.c \
-		md5-buildin/mem_clr.c \
 		$(NULL)
 CCOPT = $(V_CCOPT)
 DEFS  = $(V_DEFS)
 INCLS = $(V_INCLS)
 client_CFLAGS = $(CCOPT) $(DEFS) $(INCLS)
 client_CFLAGS += $(libcrypto_CFLAGS)
-client_LDADD  = $(LBL_LIBS)
+client_LDADD  = $(LBL_LIBS) -lcrypto
 
